lib/hawkular/inventory/inventory_api.rb

```diff
@@ -93,4 +98,24 @@ def fetch_version_and_status
       http_get('/status')
     end
   end
+
+  # Lazy fetching results, based on Inventory "ResultSet" model
+  class ResultFetcher
+    include Enumerable
+
+    def initialize(fetcher)
+      @fetcher = fetcher
+    end
+
+    def each
```

Comment:
Implementing `each` will make sure that `map` and `each` are correctly handled?

Reply:
As long as you're also including `Enumerable`, you get all those methods for free.